examples/factorial.qk

```diff
@@ -1,20 +1,11 @@
-(** vim: syntax=Scala *)
+import from console
 
--: imperative
-def fact [n] [
-  let i: int
-  let fact: int :- 1
-  for i :- 1 while i <= n do [
-    fact :- fact * i
-  ]
-  ^ fact
-]
-
--: tail_call_recursion
-def fact [n] [
-  ^ if n = 0 then 1 else [n * fact[n - 1]]
-]
-
--: compressed
-def fact! if &(= 0) then 1 else &(* fact[&0 - 1])
+fn fact(n :: number): number :- n = 1
+    then 0
+    else n * fact(n - 1)
 
+effect main
```

Comment:
I like it! :)

Reply:
This one is a really good idea. Explicitly declaring the effects sounds good. Not having this keyword will trigger errors when trying to acess external stuff?

Reply:
Yes!

Reply:
That is 10/10 <3